contracts/Bridge.sol

```diff
@@ -372,129 +246,86 @@ contract Bridge is Pausable, AccessControl, SafeMath {
-
-            emit ProposalEvent(domainID, depositNonce, ProposalStatus.Cancelled, dataHash);
-        }
+        address handler = _resourceIDToHandlerAddress[resourceID];
+        bytes32 dataHash = keccak256(abi.encodePacked(handler, data));
 
-        if (proposal._status != ProposalStatus.Cancelled) {
-            proposal._yesVotes = (proposal._yesVotes | _relayerBit(sender)).toUint200();
-            proposal._yesVotesTotal++; // TODO: check if bit counting is cheaper.
+        IDepositExecute depositHandler = IDepositExecute(handler);
 
-            emit ProposalVote(domainID, depositNonce, proposal._status, dataHash);
+        usedNonces[domainID][depositNonce/256] |= 1 << (depositNonce % 256);
 
-            // Finalize if _relayerThreshold has been reached
-            if (proposal._yesVotesTotal >= _relayerThreshold) {
-                proposal._status = ProposalStatus.Passed;
-                emit ProposalEvent(domainID, depositNonce, ProposalStatus.Passed, dataHash);
+        if (revertOnFail) {
+            depositHandler.executeProposal(resourceID, data);
+        } else {
+            try depositHandler.executeProposal(resourceID, data) {
+            } catch (bytes memory lowLevelData) {
+                emit FailedHandlerExecution(lowLevelData);
+                return;
             }
         }
-        _proposals[nonceAndID][dataHash] = proposal;
 
-        if (proposal._status == ProposalStatus.Passed) {
-            executeProposal(domainID, depositNonce, data, resourceID, false);
-        }
+        emit ProposalExecution(domainID,destinationDomainID, depositNonce, dataHash);
     }
 
     /**
-        @notice Cancels a deposit proposal that has not been executed yet.
-        @notice Only callable by relayers when Bridge is not paused.
-        @param domainID ID of chain deposit originated from.
-        @param depositNonce ID of deposited generated by origin Bridge contract.
-        @param dataHash Hash of data originally provided when deposit was made.
-        @notice Proposal must be past expiry threshold.
-        @notice Emits {ProposalEvent} event with status {Cancelled}.
+        @notice Transfers eth in the contract to the specified addresses. The parameters addrs and amounts are mapped 1-1.
+        This means that the address at index 0 for addrs will receive the amount (in WEI) from amounts at index 0.
+        @param addrs Array of addresses to transfer {amounts} to.
+        @param amounts Array of amonuts to transfer to {addrs}.
      */
-    function cancelProposal(uint8 domainID, uint64 depositNonce, bytes32 dataHash) public onlyAdminOrRelayer {
-        uint72 nonceAndID = (uint72(depositNonce) << 8) | uint72(domainID);
-        Proposal memory proposal = _proposals[nonceAndID][dataHash];
-        ProposalStatus currentStatus = proposal._status;
+    function transferFunds(address payable[] calldata addrs, uint[] calldata amounts) external onlyAdmin {
```

Comment:
Do we need this function right now since all the Fee's will be collected on the FeeHandler?

Reply:
probably a merge issue, not present in the FeeHandler version of the Bridge contract

Reply:
Yeah, seems I missed it when merging 'master'

Reply:
cleaned up